lib/ansible/cli/galaxy.py

```diff
@@ -1386,6 +1392,9 @@ def execute_list_collection(self, artifacts_manager=None):
         warnings = []
         path_found = False
         collection_found = False
+        if output_format is not None:
+            collections_in_paths = {}
```

Comment:
Declaring a variable in only one of the code branches is dangerous because it's hard to maintain accessing it under the same conditions in other places which can become a real bug during refactoring.
```suggestion
        collections_in_paths = {}
```